third-party/realdds/src/dds-device-server.cpp

```diff
@@ -135,3 +172,33 @@ void dds_device_server::publish_notification( topics::flexible_msg && notificati
     _notification_server->send_notification( std::move( notification ) );
 }
 
+
+void dds_device_server::on_control_message_received()
+{
+    topics::flexible_msg data;
+    eprosima::fastdds::dds::SampleInfo info;
+    while ( topics::flexible_msg::take_next( *_control_reader, &data, &info ) )
+    {
+        if ( !data.is_valid() )
+            continue;
+
+        _control_dispatcher.invoke( [&]( dispatcher::cancellable_timer ) { handle_control_message( data ); } );
+    }
+}
+
+void dds_device_server::handle_control_message( topics::flexible_msg control_message )
+{
+    auto const & j = control_message.json_data();
+    auto id = j["id"].get< std::string >();
```

Comment:
I think you'd be better off using the utility functions `utilities::json::get` which will take care of various other things.

Reply:
Done